cray_product_catalog/catalog_delete.py

```diff
@@ -55,13 +55,22 @@
 
 from cray_product_catalog.logging import configure_logging
 from cray_product_catalog.util import load_k8s
+from cray_product_catalog.util.catalog_data_helper import format_product_cm_name
+from cray_product_catalog.constants import (
+    CONFIG_MAP_FIELDS,
+    PRODUCT_CM_FIELDS,
+)
+
 
 urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
 LOGGER = logging.getLogger(__name__)
 
+ERR_NOT_FOUND = 404
+ERR_CONFLICT = 409
+
 
-def modify_config_map(name, namespace, product, product_version, key=None):
+def modify_config_map(name, namespace, product, product_version, max_attempts, key=None):
```

Comment:
If we want to require callers to use this new argument, then the current code is fine. If you would like to make it backwards compatible, though, you could always change it to being the final parameter and give it a default value:
```suggestion
def modify_config_map(name, namespace, product, product_version, key=None, max_attempts=100):
```